core/rest-client/src/http_client.rs

```diff
@@ -193,6 +200,14 @@ impl SendHttpRequest for HttpClient {
 
 		let mut writer = Vec::new();
 
+		if self.authenticated_connection {
+			let response = request
+				.send_with_pem_certificate(&mut writer, root_certificate)
+				.map_err(Error::HttpReqError)?;
+
+			return Ok((response, writer.clone()))
+		}
```

Comment:
I think it would be nicer to have an `if - else` here, and each branch has its own write buffer (`writer`), so you don't need a `clone()`.